tests/backup/backup_test.go

```diff
@@ -1964,9 +1960,379 @@ var _ = Describe("{ShareBackupsAndClusterWithUser}", func() {
+			}
+			wg.Wait()
+			log.Infof("The users created are %v", users)
+		})
+
+		Step("Adding roles to all the users", func() {
+			log.InfoD("Adding roles to the users")
+			for i := 0; i < numberOfUsers; i++ {
+				switch i {
+				case 0, 1, 2:
+					err = backup.AddRoleToUser(users[i], backup.ApplicationOwner, "Testing torpedo")
+					log.FailOnError(err, "Failed to add role px-backup-app.admin to user - %s", users[i])
+				case 3, 4, 5:
+					err = backup.AddRoleToUser(users[i], backup.ApplicationUser, "Testing torpedo")
+					log.FailOnError(err, "Failed to add role px-backup-app.user to user - %s", users[i])
+				case 6, 7, 8:
+					err = backup.AddRoleToUser(users[i], backup.InfrastructureOwner, "Testing torpedo")
+					log.FailOnError(err, "Failed to add role px-backup-infra.admin to user - %s", users[i])
+				default:
+					err = backup.AddRoleToUser(users[i], backup.ApplicationOwner, "Testing torpedo")
+					log.FailOnError(err, "Failed to add role px-backup-app.admin to user - %s", users[i])
+				}
+			}
+		})
+
+		Step("Creating backup location and cloud setting", func() {
+			log.InfoD("Creating backup location and cloud setting")
+			bucketNames := getBucketName()
+			providers := getProviders()
+			for _, provider := range providers {
+				bucketName := fmt.Sprintf("%s-%s", provider, bucketNames[0])
+				cloudCredName = fmt.Sprintf("%s-%s", "cred", provider)
+				bkpLocationName = fmt.Sprintf("%s-%s-bl", provider, bucketNames[0])
+				cloudCredUID = uuid.New()
+				backupLocationUID = uuid.New()
+				backupLocationMap[backupLocationUID] = bkpLocationName
+				CreateCloudCredential(provider, cloudCredName, cloudCredUID, orgID)
+				// TODO remove time.sleep: PA-509
+				time.Sleep(time.Minute * 3)
+				CreateBackupLocation(provider, bkpLocationName, backupLocationUID, cloudCredName, cloudCredUID, bucketName, orgID, "")
+			}
+		})
+		Step("Register cluster for backup", func() {
+			CreateSourceAndDestClusters(orgID, "", "", ctx)
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster")
+		})
+
+		Step("Taking backups of application for each user", func() {
+			log.InfoD("Taking backups of application for each user")
+			var sem = make(chan struct{}, 10)
+			var wg sync.WaitGroup
+			for i := 0; i < numberOfUsers; i++ {
+				sem <- struct{}{}
+				time.Sleep(3 * time.Second)
+				backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+				backupNames = append(backupNames, backupName)
+				wg.Add(1)
+				go func(backupName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					defer func() { <-sem }()
+					err = CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID, []string{bkpNamespaces[0]},
+						labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+					dash.VerifyFatal(err, nil, "Taking backup of application")
```

Comment:
Please mention the name of the backup, so that it is easier to debug by looking at the dashboard report

Reply:
done